packages/playwright-core/src/server/webkit/wkPage.ts

```diff
@@ -1157,6 +1190,12 @@ export class WKPage implements PageDelegate {
     if (!request)
       return;
 
+    // If a provisional page was created after Cross-Origin-Opener-Policy headers were received, it
+    // will take over the request (with a new id). We'll only fail it if the provisional navigation
+    // fails/get canceled.
+    if (this._provisionalPage?.pendingNavigationRequest() === request.request)
```

Comment:
It would be ideal to get rid of the extra `loadingFailed` from the original page.